data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/wrappers/CreateEnvironmentRequest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservicemodel.v1.model.wrappers;
+
+import com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentId;
+import com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentType;
+import com.amazonaws.blox.dataservicemodel.v1.model.InstanceGroup;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+@Data
+@Builder
+// required for builder
+@AllArgsConstructor
+// required for mapstruct
+@NoArgsConstructor
+public class CreateEnvironmentRequest {
+
+  @NonNull private EnvironmentId environmentId;
+  @NonNull private String role;
+  @NonNull private EnvironmentType environmentType;
+  @NonNull private String taskDefinition;
```

Comment:
At the data service level, should we perhaps require that this has been resolved to a full ARN? In general, I think it might make things simpler if all IDs passed into the data service were ARNs already, so that we don't have to deal with maybe-arn-maybe-name semantics beyond this interface.

Reply:
what's an id here? environmentid? the taskdef and role are arns.

Reply:
Specifically the identifiers for entities that aren't owned by blox, in this case. So `role`, `taskDefinition` and `cluster`. Maybe? Maybe it's more a documentation/implementation detail. Not a blocker for this CR, but worth keeping in mind for implementation.